tensorflow_probability/python/experimental/mcmc/covariance_reducer.py

```diff
@@ -61,46 +67,92 @@ def __init__(self, shape, event_ndims=None, dtype=tf.float32, name=None):
     """Initializes a `RunningCovarianceState` using previously defined metadata.
 
+    For calculation purposes, the `initial_chain_state` does not count as a
+    sample. This is a deliberate decision that ensures consistency across
+    sampling procedures (i.e. `tfp.mcmc.sample_chain` follows similar
+    semantics).
+
     Args:
-      initial_chain_state: `Tensor` or Python `list` of `Tensor`s representing
-        the current state(s) of the Markov chain(s). For streaming covariance,
-        this argument has no influence on computation. Hence, by default, it
-        is `None`.
+      initial_chain_state: A (possibly nested) structure of `Tensor`s or Python
+        `list`s of `Tensor`s representing the current state(s) of the Markov
+        chain(s). For streaming covariance, this argument has no influence on
+        computation. Hence, by default, it is `None`. However, this argument is
+        still accepted as it will be supplied by the Streaming MCMC framework
+        and is used to infer the shape and dtype of future samples.
       initial_kernel_results: A (possibly nested) `tuple`, `namedtuple` or
         `list` of `Tensor`s representing internal calculations made in a related
         `TransitionKernel`. For streaming covariance, this argument also has no
-        influence on computation; hence, it is also `None` by default.
+        influence on computation; hence, it is also `None` by default. Likewise,
+        this argument is still accepted as it will be supplied by the Streaming
+        MCMC framework.
 
     Returns:
-      state: `RunningCovarianceState` representing a stream of no inputs.
+      state: `RunningCovarianceState` representing a stream of no inputs and
+        with identical structure to the `initial_chain_state`.
     """
-    # `initial_chain_state` not included as a sample for consistency
-    # with `sample_chain`
     with tf.name_scope(
         mcmc_util.make_name(self.name, 'covariance_reducer', 'initialize')):
-      return self.strm.initialize()
+      # pylint: disable=unnecessary-lambda
+      initial_chain_state = tf.nest.map_structure(
+          lambda initial_chain_state: tf.convert_to_tensor(initial_chain_state),
+          initial_chain_state)
+      self._parameters['shape'] = tf.nest.map_structure(
+          lambda chain_state: chain_state.shape,
+          initial_chain_state)
+      self._parameters['dtype'] = tf.nest.map_structure(
+          lambda chain_state: chain_state.dtype,
+          initial_chain_state)
+      if self.event_ndims is None:
+        self._parameters['event_ndims'] = tf.nest.map_structure(
+            lambda chain_state: ps.rank(chain_state),
+            initial_chain_state
+        )
+      elif not nest.is_sequence(self.event_ndims):
+        self._parameters['event_ndims'] = tf.nest.map_structure(
+            lambda _: self.event_ndims,
+            initial_chain_state,
+        )
+      self.strm = nest.map_structure_up_to(
```

Comment:
Probably better to name self.strm in the plural, since the rest of the code will treat it as a collection of streaming computations, rather than just one.